css/css-content/content-animation.html

```diff
@@ -23,29 +27,39 @@
 
 </style>
 <body>
-<div class="target"></div>
 <script>
 
-test(function() {
-  const target = document.querySelector(".target");
-  const style = getComputedStyle(target, "::after");
+const contentAnimationTest = (className, expectedInitialValue, title) => {
+    test(() => {
+        const target = document.body.appendChild(document.createElement("div"));
+        target.classList.add("target");
+        if (className)
+            target.classList.add(className);
+
+        const style = getComputedStyle(target, "::after");
 
-  assert_equals(style.content, '"default"', "Before the animation is applied.");
+        assert_equals(style.content, expectedInitialValue, "Before the animation is applied.");
 
-  target.classList.add("animated");
-  const animation = target.getAnimations({ subtree: true })[0];
+        target.classList.add("animated");
+        const animation = target.getAnimations({ subtree: true })[0];
 
-  const testContentAtTime = (time, expected) => {
-    animation.currentTime = time;
-    assert_equals(style.content, `"${expected}"`, `Check the animated value at time = ${time}ms`);
-  };
+        const testContentAtTime = (time, expected) => {
+          animation.currentTime = time;
+          assert_equals(style.content, `"${expected}"`, `Check the animated value at time = ${time}ms`);
+        };
+
+        testContentAtTime(0, 'from');
+        testContentAtTime(499, 'from');
+        testContentAtTime(500, 'to');
+        testContentAtTime(999, 'to');
+        testContentAtTime(1000, 'to');
+
+        target.remove();
+    }, title);
+}
 
-  testContentAtTime(0, 'from');
-  testContentAtTime(499, 'from');
-  testContentAtTime(500, 'to');
-  testContentAtTime(999, 'to');
-  testContentAtTime(1000, 'to');
-}, "The content property can be animated with a discrete animation type.");
+contentAnimationTest('', 'none', 'The content property can be animated with a discrete animation type when there is no initial content value.');
+contentAnimationTest('initial', '"initial"', 'The content property can be animated with a discrete animation type when there is an initial content value.');
```

Comment:
This really confused me, I thought this was the initial CSS wide keyword (which should resolve to `none`), can we use something else? Like "before-animation" or "without-animation"

Reply:
Also, can we make the 2 first parameters an object? Typically when I see two parameters like this, I think: parameter 1 = specified style, parameter 2 = resolved/computed style.

Something like this would be more clear:
```
test_content_animation({
  className: "string-content",
  valueBeforeAnimation: `"not-animated"`
}, "The content property can be animated with a discrete animation type when there is an initial content value.");
```

You can also omit the first `className` parameter this way, for the no class name case.